internal/controllers/gateway/gateway_controller.go

```diff
@@ -266,6 +276,43 @@ func (r *GatewayReconciler) listGatewaysForService(svc client.Object) (recs []re
 	return
 }
 
+// listGatewaysForHTTPRoute retrieves all the gateways referenced as parents by the HTTPRoute.
+func (r *GatewayReconciler) listGatewaysForHTTPRoute(obj client.Object) []reconcile.Request {
+	httpRoute, ok := obj.(*gatewayv1beta1.HTTPRoute)
+	if !ok {
+		r.Log.Error(
+			fmt.Errorf("unexpected object type"),
+			"httproute watch predicate received unexpected object type",
+			"expected", "*gatewayv1beta1.HTTPRoute", "found", reflect.TypeOf(obj),
+		)
+		return nil
+	}
+	recs := []reconcile.Request{}
+	for _, routeParentStatus := range httpRoute.Status.Parents {
+		gatewayNamespace := httpRoute.Namespace
+		parentRef := routeParentStatus.ParentRef
+		if (parentRef.Group != nil && *parentRef.Group != gatewayV1beta1Group) ||
+			(parentRef.Kind != nil && *parentRef.Kind != "Gateway") {
+			continue
+		}
+		if parentRef.Namespace != nil {
+			gatewayNamespace = string(*parentRef.Namespace)
+		}
+		if lo.ContainsBy(routeParentStatus.Conditions, func(condition metav1.Condition) bool {
+			return condition.Type == string(gatewayv1beta1.RouteConditionAccepted) &&
+				condition.Status == metav1.ConditionTrue
+		}) {
```

Comment:
First I just wanna say I'm not necessarily against using the `lo` package, but I am curious as to the growing preference for it, could you help me to better understand?

At least in this case I feel like "does a provided list of conditions include a specific condition marked true?" might be a generic enough concept that a simple utility function for it could work here:

```suggestion
		if utils.ConditionListIncludesConditionTrue(routeParentStatus.Conditions, gatewayv1beta1.RouteConditionAccepted) {
```

I think one could argue this is slightly clearer at a glance, and also reduces clutter in this method, but it's a bit of a nit. I am curious to hear your thoughts? :thinking:.

Reply:
From my PoV: I'm somewhere in the middle between advocating for usage of `lo` and using special purpose functions.

I'd be inclined to use the other if and only if we could write those in a way that doesn't make them to have only 1 call site throughout the code base. So either add a parameter or a type parameter and try to  use it in other places where it can shave off several lines of code.

Reply:
> I'd be inclined to use the other if and only if we could write those in a way that doesn't make them to have only 1 call site throughout the code base

100% agree. I think introducing such a helper here is a bit out of scope. We could explore the code base to see how massively we are using `lo` and in what cases creating a helper could be a better fit.